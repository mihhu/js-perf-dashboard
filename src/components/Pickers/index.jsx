import React from 'react';
import PropTypes from 'prop-types';
import { withStyles } from '@material-ui/core/styles';
import Picker from '../Picker';
import { BENCHMARKS, CONFIG } from '../../config';

const styles = () => ({
  root: {
    display: 'flex',
    textAlign: 'center',
    padding: '15px',
  },
});

const Pickers = ({
  classes, benchmark, onChange, platform,
}) => (
  <div className={classes.root}>
    <Picker
      key="Platform selection"
      identifier="platform"
      topLabel="Platform"
      onSelection={onChange}
      selectedValue={platform}
      options={
        Object.keys(CONFIG.platforms).reduce((res, platformKey) => {
          res.push({ value: platformKey, label: CONFIG.platforms[platformKey].label });
          return res;
        }, [])
      }
    />
    <Picker
      key="Benchmark selection"
      identifier="benchmark"
      topLabel="Benchmark"
      onSelection={onChange}
      selectedValue={benchmark}
      options={
        CONFIG.platforms[platform].benchmarks.sort().reduce((res, benchmarkKey) => {
          res.push({
            value: benchmarkKey,
            label: BENCHMARKS[benchmarkKey].label,
          });
          return res;
        }, [{ value: 'overview', label: 'Overview' }])
      }
    />
  </div>
);

Pickers.propTypes = ({
  classes: PropTypes.shape().isRequired,
  benchmark: PropTypes.string.isRequired,
  onChange: PropTypes.func.isRequired,
  platform: PropTypes.string.isRequired,
});

export default withStyles(styles)(Pickers);
