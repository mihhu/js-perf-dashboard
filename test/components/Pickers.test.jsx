import React from 'react';
import renderer from 'react-test-renderer';
import 'raf/polyfill';
import Pickers from '../../src/components/Pickers';

it('renders correctly', () => {
  const tree = renderer
    .create((
      <Pickers
        platform="win10"
        benchmark="overview"
        onChange={() => {}}
      />
    ))
    .toJSON();
  expect(tree).toMatchSnapshot();
});
