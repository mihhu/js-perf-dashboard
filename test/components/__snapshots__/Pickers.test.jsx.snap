// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`renders correctly 1`] = `
<div
  className="Pickers-root-1"
>
  <form
    autoComplete="off"
    className="Picker-root-2"
  >
    <div
      className="MuiFormControl-root-3"
    >
      <label
        className="MuiFormLabel-root-14 MuiFormLabel-filled-18 MuiInputLabel-root-7 MuiInputLabel-formControl-8 MuiInputLabel-animated-11 MuiInputLabel-shrink-10"
        data-shrink={true}
      >
        Platform
      </label>
      <div
        className="MuiInputBase-root-41 MuiInput-root-28 MuiInput-underline-32 MuiInputBase-formControl-42 MuiInput-formControl-29"
        onClick={[Function]}
      >
        <div
          className="MuiSelect-root-21"
        >
          <div
            aria-haspopup="true"
            aria-owns={null}
            aria-pressed="false"
            className="MuiSelect-select-22 MuiSelect-selectMenu-25 MuiInputBase-input-51 MuiInput-input-36"
            onBlur={[Function]}
            onClick={[Function]}
            onFocus={[Function]}
            onKeyDown={[Function]}
            role="button"
            tabIndex={0}
          >
            Windows 10 64bit
          </div>
          <input
            name="platform"
            type="hidden"
            value="win10"
          />
          <svg
            aria-hidden="true"
            className="MuiSvgIcon-root-58 MuiSelect-icon-27"
            focusable="false"
            role="presentation"
            viewBox="0 0 24 24"
          >
            <path
              d="M7 10l5 5 5-5z"
            />
          </svg>
        </div>
      </div>
    </div>
  </form>
  <form
    autoComplete="off"
    className="Picker-root-2"
  >
    <div
      className="MuiFormControl-root-3"
    >
      <label
        className="MuiFormLabel-root-14 MuiFormLabel-filled-18 MuiInputLabel-root-7 MuiInputLabel-formControl-8 MuiInputLabel-animated-11 MuiInputLabel-shrink-10"
        data-shrink={true}
      >
        Time range
      </label>
      <div
        className="MuiInputBase-root-41 MuiInput-root-28 MuiInput-underline-32 MuiInputBase-formControl-42 MuiInput-formControl-29"
        onClick={[Function]}
      >
        <div
          className="MuiSelect-root-21"
        >
          <div
            aria-haspopup="true"
            aria-owns={null}
            aria-pressed="false"
            className="MuiSelect-select-22 MuiSelect-selectMenu-25 MuiInputBase-input-51 MuiInput-input-36"
            onBlur={[Function]}
            onClick={[Function]}
            onFocus={[Function]}
            onKeyDown={[Function]}
            role="button"
            tabIndex={0}
          >
            Last 90 days
          </div>
          <input
            name="timeRange"
            type="hidden"
            value="90"
          />
          <svg
            aria-hidden="true"
            className="MuiSvgIcon-root-58 MuiSelect-icon-27"
            focusable="false"
            role="presentation"
            viewBox="0 0 24 24"
          >
            <path
              d="M7 10l5 5 5-5z"
            />
          </svg>
        </div>
      </div>
    </div>
  </form>
  <form
    autoComplete="off"
    className="Picker-root-2"
  >
    <div
      className="MuiFormControl-root-3"
    >
      <label
        className="MuiFormLabel-root-14 MuiFormLabel-filled-18 MuiInputLabel-root-7 MuiInputLabel-formControl-8 MuiInputLabel-animated-11 MuiInputLabel-shrink-10"
        data-shrink={true}
      >
        Benchmark
      </label>
      <div
        className="MuiInputBase-root-41 MuiInput-root-28 MuiInput-underline-32 MuiInputBase-formControl-42 MuiInput-formControl-29"
        onClick={[Function]}
      >
        <div
          className="MuiSelect-root-21"
        >
          <div
            aria-haspopup="true"
            aria-owns={null}
            aria-pressed="false"
            className="MuiSelect-select-22 MuiSelect-selectMenu-25 MuiInputBase-input-51 MuiInput-input-36"
            onBlur={[Function]}
            onClick={[Function]}
            onFocus={[Function]}
            onKeyDown={[Function]}
            role="button"
            tabIndex={0}
          >
            Overview
          </div>
          <input
            name="benchmark"
            type="hidden"
            value="overview"
          />
          <svg
            aria-hidden="true"
            className="MuiSvgIcon-root-58 MuiSelect-icon-27"
            focusable="false"
            role="presentation"
            viewBox="0 0 24 24"
          >
            <path
              d="M7 10l5 5 5-5z"
            />
          </svg>
        </div>
      </div>
    </div>
  </form>
</div>
`;
